pgbedrock/attributes.py

```diff
@@ -271,9 +273,14 @@ def set_all_attributes(self, attributes):
         """ Verify that the role's attributes match the spec's, updating as necessary """
         for attribute, desired_value in attributes.items():
             current_value = self.get_attribute_value(attribute)
-            if attribute == 'rolpassword' and not self.is_same_password(desired_value):
-                logger.debug('Altering password for role "{}"'.format(self.rolename))
-                self.set_password(desired_value)
+            if attribute == 'rolpassword':
+                if self.is_google_cloud:
+                    logger.debug('Skipping password change on Google Cloud SQL')
```

Comment:
I personally would like this to be a warning

Reply:
I don’t know. It would be the only place that emits a warning, it breaks progress output, and is printed every time pgbedrock runs (even when there are no changes)

```
Analyzing roles:        [------------------------------------]    2%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#-----------------------------------]    5%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [##----------------------------------]    7%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [###---------------------------------]   10%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [####--------------------------------]   13%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#####-------------------------------]   15%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [######------------------------------]   18%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#######-----------------------------]   21%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [########----------------------------]   23%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#########---------------------------]   26%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [##########--------------------------]   28%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [###########-------------------------]   31%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [############------------------------]   34%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#############-----------------------]   36%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [##############----------------------]   39%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [###############---------------------]   42%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [################--------------------]   44%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#################-------------------]   47%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [##################------------------]   50%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [##################------------------]   52%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [###################-----------------]   55%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [####################----------------]   57%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#####################---------------]   60%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [######################--------------]   63%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#######################-------------]   65%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [########################------------]   68%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#########################-----------]   71%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [##########################----------]   73%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [###########################---------]   76%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [############################--------]   78%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#############################-------]   81%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [##############################------]   84%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [###############################-----]   86%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [################################----]   89%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [#################################---]   92%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [##################################--]   94%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [###################################-]   97%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [####################################]  100%  [REDACTED] WARNING:attributes.py:set_all_attributes:278 - Skipping password change on Google Cloud SQL
Analyzing roles:        [####################################]  100%
Analyzing memberships:  [####################################]  100%
Analyzing ownerships:   [####################################]  100%
Analyzing privileges:   [####################################]  100%

No changes needed. Congratulations! :)
```

Reply:
In `core_configure.py::configure()`, after `cursor = db_connection.cursor(cursor_factory=psycopg2.extras.DictCursor)` you could add a line like `dbcontext = DatabaseContext(cursor, verbose)` and then do `if dbcontext.get_version_info().is_google_cloud: click.secho(...)`

Considering we push towards a self-service model where we ask devs to look at the CI logs carefully, giving them good hints, excellent logs would be great. If that doesn't clutter the logs to much ofc.

Reply:
Yep, makes sense. I’ve added that:

```
Running on Google Cloud SQL, password changes will be skipped.
Analyzing roles:        [####################################]  100%
Analyzing memberships:  [####################################]  100%
Analyzing ownerships:   [####################################]  100%
Analyzing privileges:   [####################################]  100%

No changes needed. Congratulations! :)
```